pkg/relayer/proxy/relay_builders.go

```diff
@@ -0,0 +1,59 @@
+package proxy
+
+import (
+	"io"
+	"net/http"
+
+	"pocket/x/service/types"
+	sessiontypes "pocket/x/session/types"
+)
+
+// newRelayRequest builds a RelayRequest from an http.Request.
+func (j *jsonRPCServer) newRelayRequest(request *http.Request) (*types.RelayRequest, error) {
+	requestBz, err := io.ReadAll(request.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var relayRequest types.RelayRequest
+	if err := relayRequest.Unmarshal(requestBz); err != nil {
+		return nil, err
+	}
+
+	return &relayRequest, nil
+}
+
+// newRelayResponse builds a RelayResponse from an http.Response and a SessionHeader.
+// It also signs the RelayResponse and assigns it to RelayResponse.Meta.SupplierSignature.
+// If the response has a non-nil body, it will be parsed as a JSONRPCResponsePayload.
+func (j *jsonRPCServer) newRelayResponse(
+	response *http.Response,
+	sessionHeader *sessiontypes.SessionHeader,
+) (*types.RelayResponse, error) {
+	relayResponse := &types.RelayResponse{
+		Meta: &types.RelayResponseMetadata{SessionHeader: sessionHeader},
+	}
+
+	if response.Body != nil {
```

Comment:
If it's nil, should we just return an error and exit quickly?

Reply:
It is safe to just remove the check as `io.ReadAll(nil)` would return an empty []byte